tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcExpr.scala

```diff
@@ -17,6 +17,25 @@ sealed trait EtcExpr {
    * The reference to the source. This identifier is not taken into account in equals and hash.
    */
   val sourceRef: EtcRef
+
+  /**
+   * A function to explain type errors for this expression in case they occur.
+   */
+  var typeErrorExplanation: (List[TlaType1], List[TlaType1]) => String = (_, _) => ""
```

Comment:
```suggestion
  var typeErrorExplanation: (List[TlaType1], List[TlaType1]) => Option[String] = (_, _) => None
```

Is there an advantage to `""` as default value?

Reply:
The initial idea was to make it easy to interpolate into other error messages as a complement without having to perform checks. But since we are now replacing the whole message, using optional types seems better.